homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -298,8 +298,16 @@ def device_class(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes of the sensor."""
+
+        attr_observations_list = list(
+            obs.copy() for obs in self.current_observations.values() if obs is not None
```

Comment:
I wonder if the bug here is that `value_template` is in `current_observations` to begin with.

Reply:
the `value_template` is needed here: https://github.com/home-assistant/core/blob/50e7b4dc10d40bcee90f243468ee57e9413f2e0d/homeassistant/components/bayesian/binary_sensor.py#L235

I believe the bug was introduced because I changed the data structures to yield some simpler logic, and I didn't have a general awareness to check to make sure that any data structure exposed as a state attribute needs to be json serializable.

Reply:
actually -- hmm digging a bit further

Reply:
So that line referenced isn't exactly how it's used, but it _is_ essential.  Each handler defined here (https://github.com/home-assistant/core/blob/50e7b4dc10d40bcee90f243468ee57e9413f2e0d/homeassistant/components/bayesian/binary_sensor.py#L133) takes an "observation" dictionary that it can process.  In the case of the template processor, a `value_template` must exist in that "observation".

These "observation" dictionaries were then being exposed directly via the `current_observations` attribute.

Reply:
Seems to resolve my issue with hanging at "loading data".  Will continue testing, thanks.